contributing/metadata.md

```diff
@@ -0,0 +1,166 @@
+
+```shell
+yarn update-registry
+```
+
+### What the script can't do
+
+inFolderTypes and types with childXml in their describe are not supported. You **want** to explore the various strategies for those (see the SDR Handbook in this repo) and create NUTs to validate that your types are being handled correctly.
+
+For those situations, refer to another existing type in the registry that you want yours to behave like.
+
+If that's confusing, it's a great time to reach out to the CLI team.
+
+## Manual Edits
+
+You can do what the script does yourself. As you work, run `./node_modules/mocha/bin/mocha test/registry/registryValidation.test.ts` to check your entries
+
+Run `sfdx force:mdapi:describemetadata --json` to get the describe. `>` it to a file or use `|jq` to find your type.
+
+Your describe will contains something like this
+
+```json
+{
+    "directoryName": "yourTypes",
+    "inFolder": "false",
+    "metaFile": "false",
+    "suffix": "yourtype",
+    "xmlName": "YourNewType"
+},
+```
+
+Entries in the describe.json have the following shape:
+| Property | Description|
+| --- | --- |
+|childXmlNames | List of child sub-components for this component|
+|directoryName | The name of the directory in the .zip file that contains this component.|
+| inFolder | Indicates whether the component is in a folder (true) or not (false). For example, documents, email templates and reports are stored in folders.|
+|metaFile| Indicates whether the component requires an accompanying metadata file. For example, documents, classes, and s-controls are components that require an additional metadata file.|
+|suffix| The file suffix for this component.|
+|xmlName| The name of the root element in the metadata file for this component. This name also appears in the Packages > types > name field in the manifest file package.xml.|
+
+You'd create an entry in the `types` property.
+
+> this fictitious type COULD HAVE used the automated script
+
+```json
+"yournewtype": { // the key is lowercased xmlName
+      "id": "yournewtype", // lowercased xmlName
+      "name": "YourNewType", // xmlName
+      "suffix": "yourtype", // suffix
+      "directoryName": "yourTypes", //directoryName
+      "inFolder": false, // inFolder
+      "strictDirectoryName": false // set to true when suffixes are not unique in the registry
+    },
+```
+
+add a matching entry in `suffixes`
+
+```json
+  "yourtype": "yournewtype" // "suffix": "id"
+```
+
+### When suffixes aren't unique
+
+If your type uses a suffix already in use, you'll need to tell SDR to use the directoryName as well by setting `"strictDirectoryName"` : `"true"`
+
+Then also add an entry in the `strictDirectoryNames` section.
+
+```json
+  "yourtype" : "yournewtype" // "directory" : "id"
+
+```
+
+To preserve existing behavior, use `strictDirectoryName` on the new types, not the old ones.
+
+# Testing
+
+SDR include 2 registry-related tests to check your changes to the metdataRegistry.json
```

Comment:
```suggestion
SDR includes 2 registry-related tests to check your changes against the `metdataRegistry.json`
```

:warning: Confirm that the word "against" is appropriate. "Check your changes _to_" was awkward to read.